internal/store/db/redis/store.go

```diff
@@ -16,37 +16,212 @@
-	return nil, nil
+func (c Client) RetrieveFromStore(appServiceKey string) (objects []contracts.StoredObject, err error) {
+	// do not satisfy requests for a blank ASK
+	if appServiceKey == "" {
+		return nil, errors.New("no AppServiceKey provided")
+	}
+
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	ids, err := redis.Values(conn.Do("SMEMBERS", appServiceKey))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ids) == 0 {
+		return nil, nil
+	}
+
+	values, err := redis.ByteSlices(conn.Do("MGET", ids...))
+	if err != nil {
+		return nil, err
+	}
+
+	var model models.StoredObject
+
+	for _, bytes := range values {
+		err = model.UnmarshalJSON(bytes)
+		if err != nil {
+			return nil, err
+		}
+		objects = append(objects, model.ToContract())
+	}
+
+	return objects, nil
 }
 
 // Update replaces the data currently in the store with the provided data.
-func Update(o contracts.StoredObject) error {
+func (c Client) Update(o contracts.StoredObject) error {
+	err := o.ValidateContract(true)
+	if err != nil {
+		return err
+	}
+
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	err = removeFromDB(conn, o)
+	if err != nil {
+		return err
+	}
+
+	_, err = addToDB(conn, o)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // RemoveFromStore removes an object from the data store.
-func RemoveFromStore(id string) error {
-	return nil
+func (c Client) RemoveFromStore(o contracts.StoredObject) error {
+	err := o.ValidateContract(true)
+	if err != nil {
+		return err
+	}
+
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	return removeFromDB(conn, o)
 }
 
 // Disconnect ends the connection.
-func Disconnect() error {
-	return nil
+func (c Client) Disconnect() error {
+	return c.Pool.Close()
 }
 
 // NewClient provides a factory for building a StoreClient
 func NewClient(config db.DatabaseInfo) (interfaces.StoreClient, error) {
-	return nil, nil
+	once.Do(func() {
+		connectionString := fmt.Sprintf("%s:%d", config.Host, config.Port)
+		opts := []redis.DialOption{
+			redis.DialPassword(config.Password),
+			redis.DialConnectTimeout(time.Duration(config.Timeout) * time.Millisecond),
+		}
+
+		dialFunc := func() (redis.Conn, error) {
+			conn, err := redis.Dial(
+				"tcp", connectionString, opts...,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("Could not dial Redis: %s", err)
+			}
+			return conn, nil
+		}
+		currClient = &Client{
+			Pool: &redis.Pool{
+				IdleTimeout: 0,
+				/* The current implementation processes nested structs using concurrent connections.
+				 * With the deepest nesting level being 3, three shall be the number of maximum open
+				 * idle connections in the pool, to allow reuse.
+				 * TODO: Once we have a concurrent benchmark, this should be revisited.
+				 * TODO: Longer term, once the objects are clean of external dependencies, the use
+				 * of another serializer should make this moot.
+				 */
+				MaxIdle: 10,
+				Dial:    dialFunc,
+			},
+			BatchSize: 1000,
```

Comment:
Should we make hard coded values constants?

Reply:
I will readd BatchSize and the other removed Redis only constants to the configuration.